src/Avalonia.Visuals/Media/Rotate3DTransform.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Numerics;
+
+namespace Avalonia.Media;
+
+/// <summary>
+///  Non-Affine 3D transformation for rotating an visual around a definable axis
+/// </summary>
+public class Rotate3DTransform : Transform
+{
+    /// <summary>
+    /// Defines the <see cref="RotationX"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> RotationXProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(RotationX));
+
+    /// <summary>
+    /// Defines the <see cref="RotationY"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> RotationYProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(RotationY));
+
+    /// <summary>
+    /// Defines the <see cref="RotationZ"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> RotationZProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(RotationZ));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterX"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterXProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterX));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterY"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterYProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterY));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterZ"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterZProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterZ));
+
+    /// <summary>
+    /// Defines the <see cref="Depth"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> DepthProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(Depth));
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Rotate3DTransform"/> class.
+    /// </summary>
+    public Rotate3DTransform()
+    {
+        this.GetObservable(RotationXProperty).Subscribe(_ => RaiseChanged());
```

Comment:
IIRC `GetObservable` is quite slow. You should be using `protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)` and check for the properties in there.

Reply:
> For issue 2:
> 
> > Invalidation of the area to be redrawn seems to be broken. A lot of artefacts remain while the element is rotating.
> 
> Maybe this issue is related? [animationartifacts](https://github.com/AvaloniaUI/Avalonia/issues/5381)

Thank you, the artifacts look similar, solution is the same (resizing the window). Also noticed this on other controls, like the thumb of the slider which sometimes leaves a trace on the screen when dragged.